### PR TITLE
Add a subtle animation to the 'duplicate' function #863

### LINKED ISSUE #863: Add a subtle animation to the 'duplicate' function
May be duplicated object can be put a few pixels off from the original location and then automatically moved to the original location (animated) so that the user sees the duplicate was created?

### PR DESCRIPTION
**Pull request by vhazali**
Fixes #863

### CODE REVIEW

File: PowerPointLabs/PowerPointLabs/Models/PowerPointSlide.cs [hunk 1]
```diff
@@ -560,7 +560,7 @@ public Shape CopyShapeToSlide(Shape shape)
         /// If you are having difficulty getting a shaperange, use the ToShapeRange method.
         /// TODO: Test this method more thoroughly in more cases other than Graphics.SquashSlides
         /// </summary>
-        public ShapeRange CopyShapesToSlide(ShapeRange shapes)
+        public ShapeRange CopyShapesToSlide(ShapeRange shapes, Boolean needOffset)
```
Comment (damithc) [inline PowerPointLabs/PowerPointLabs/Models/PowerPointSlide.cs]: `needsOffset`?

Comment (vhazali) [inline PowerPointLabs/PowerPointLabs/Models/PowerPointSlide.cs]: oops. Edited!

File: PowerPointLabs/PowerPointLabs/Models/PowerPointSlide.cs [hunk 2]
```diff
@@ -560,7 +560,7 @@ public Shape CopyShapeToSlide(Shape shape)
         /// If you are having difficulty getting a shaperange, use the ToShapeRange method.
         /// TODO: Test this method more thoroughly in more cases other than Graphics.SquashSlides
         /// </summary>
-        public ShapeRange CopyShapesToSlide(ShapeRange shapes)
+        public ShapeRange CopyShapesToSlide(ShapeRange shapes, Boolean needsOffset)
```
Comment (kai33) [inline PowerPointLabs/PowerPointLabs/Models/PowerPointSlide.cs]: Turn it into private if possible

**Comment by kai33** [comment]
Pls write unit test to assert your changes. There's a readme under Test project telling how to write unit test.

File: PowerPointLabs/PowerPointLabs/Models/PowerPointSlide.cs (hunk 2 above)
Comment (vhazali) [inline PowerPointLabs/PowerPointLabs/Models/PowerPointSlide.cs]: Got it

**Comment by Ohohcakester** [comment]
I am against this.
The whole purpose of the duplicate function is to duplicate without an offset.
This is one of the most difficult things to do in powerpoint.
You have to add the feedback a different way.

**Comment by damithc** [comment]
The intention of the issue is not to offset the duplicate. What I wanted was to create a subtle animation that serves as a visual cue to indicate that the duplicate was created. The animation I suggested was to put the duplicate slightly off from the original position and then automatically move it to the original position in a few steps (0.2 sec per step?).

**Comment by vhazali** [comment]
I see. It seems I have misunderstood the issue. I'll find another way to do the visual cue

**Comment by Ohohcakester** [comment]
It would be good to read what was written in the comments before making changes. It pays to be a little more careful.

Also, avoid using Boolean in C#. The primitive type for boolean variables is "bool". Unlike java, wrappers for primitive objects, like Integer, Float, Boolean, are almost never used. (e.g. var foo = new List<int>() )

**Comment by vhazali** [comment]
Yup, sorry 'bout the confusion. 
Will change the type to bool also. Thanks!

**Pull request closed without merging**